app/src/main/java/org/gdg/frisbee/android/app/OrganizerChecker.java

```diff
@@ -45,40 +43,44 @@ public boolean isOrganizer() {
         return isOrganizer;
     }
 
-    void checkOrganizer(GoogleApiClient apiClient, final Callbacks responseHandler) {
-        if (!PrefUtils.isSignedIn(apiClient.getContext())) {
+    void checkOrganizer(Context context, final Callbacks responseHandler) {
+        final String currentId = PlusUtils.getCurrentPersonId(context);
+
+        if (currentId == null) {
             isOrganizer = false;
             checkedId = null;
             responseHandler.onOrganizerResponse(false);
             return;
         }
 
-        final String currentId = PlusUtils.getCurrentPersonId(apiClient);
-
-        if (currentId != null
-            && (!currentId.equals(checkedId)
-            || System.currentTimeMillis() > lastOrganizerCheckTimeStamp + ORGANIZER_CHECK_MAX_TIME)) {
-            isOrganizer = false;
-            App.getInstance().getGdgXHub().checkOrganizer(currentId).enqueue(new Callback<OrganizerCheckResponse>() {
-                @Override
-                public void success(OrganizerCheckResponse organizerCheckResponse) {
-                    lastOrganizerCheckTimeStamp = System.currentTimeMillis();
-                    checkedId = currentId;
-                    isOrganizer = organizerCheckResponse.getChapters().size() > 0;
-                    responseHandler.onOrganizerResponse(isOrganizer);
-
-                    savePreferences();
-                }
-
-                @Override
-                public void failure(Throwable error) {
-                    isOrganizer = false;
-                    responseHandler.onErrorResponse();
-                }
-            });
-        } else {
+        if (isLastOrganizerCheckValid(currentId)) {
             responseHandler.onOrganizerResponse(isOrganizer);
+            return;
         }
+
+        isOrganizer = false;
+        App.getInstance().getGdgXHub().checkOrganizer(currentId).enqueue(new Callback<OrganizerCheckResponse>() {
+            @Override
+            public void success(OrganizerCheckResponse organizerCheckResponse) {
+                lastOrganizerCheckTimeStamp = System.currentTimeMillis();
+                checkedId = currentId;
+                isOrganizer = organizerCheckResponse.getChapters().size() > 0;
+                responseHandler.onOrganizerResponse(isOrganizer);
+
+                savePreferences();
+            }
+
+            @Override
+            public void failure(Throwable error) {
+                isOrganizer = false;
+                responseHandler.onErrorResponse();
+            }
+        });
+    }
+
+    private boolean isLastOrganizerCheckValid(String currentId) {
```

Comment:
This method name is a bit strange.. just looking at the code and the name I'm not sure what we're doing here.